components/timezone/src/zoned_datetime.rs

```diff
@@ -4,35 +4,108 @@
 
 use icu_calendar::{AsCalendar, Date, Iso, Time};
 
-use crate::CustomTimeZone;
+use crate::{FormattableTimeZone, TimeZone};
+use crate::{MetazoneCalculator, ZoneOffsetCalculator};
 
 /// A date and time local to a specified custom time zone.
 #[derive(Debug)]
 #[allow(clippy::exhaustive_structs)] // this type is stable
-pub struct CustomZonedDateTime<A: AsCalendar> {
+pub struct ZonedDateTime<A: AsCalendar> {
     /// The date, local to the time zone
     pub date: Date<A>,
     /// The time, local to the time zone
     pub time: Time,
     /// The time zone
-    pub zone: CustomTimeZone,
+    pub zone: TimeZone,
 }
 
-impl<A: AsCalendar> CustomZonedDateTime<A> {
-    /// Convert the CustomZonedDateTime to an ISO CustomZonedDateTime
+impl<A: AsCalendar> ZonedDateTime<A> {
+    /// Convert the ZonedDateTime to an ISO ZonedDateTime
     #[inline]
-    pub fn to_iso(&self) -> CustomZonedDateTime<Iso> {
-        CustomZonedDateTime {
+    pub fn to_iso(&self) -> ZonedDateTime<Iso> {
+        ZonedDateTime {
             date: self.date.to_iso(),
             time: self.time,
             zone: self.zone,
         }
     }
 
-    /// Convert the CustomZonedDateTime to a CustomZonedDateTime in a different calendar
+    /// Convert the ZonedDateTime to a ZonedDateTime in a different calendar
     #[inline]
-    pub fn to_calendar<A2: AsCalendar>(&self, calendar: A2) -> CustomZonedDateTime<A2> {
-        CustomZonedDateTime {
+    pub fn to_calendar<A2: AsCalendar>(&self, calendar: A2) -> ZonedDateTime<A2> {
+        ZonedDateTime {
+            date: self.date.to_calendar(calendar),
+            time: self.time,
+            zone: self.zone,
+        }
+    }
+
+    /// Convert the [`ZonedDateTime`] to a [`FormattableZonedDateTime`].
+    pub fn to_formattable(
+        self,
+        metazone_calculator: &MetazoneCalculator,
+        zone_offset_calculator: &ZoneOffsetCalculator,
```

Comment:
This feels obvious, but we should probably try to merge these down to one type. I can't think of any use case where someone would want to load a MetazoneCalculator and a ZoneOffsetCalculator from different data sources.

Reply:
I think for now just keep `MetazoneCalculator` and add the zone offset data marker to its signature. It will load two data payloads. Eventually we should merge the data payloads into one.